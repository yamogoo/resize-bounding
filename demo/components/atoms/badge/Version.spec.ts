import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";

import Version from "./Version.vue";

const DEFAULT_PROPS = {
  version: "0.0.1",
};

describe("Version", () => {
  test(`should display "version" (${DEFAULT_PROPS.version})`, () => {
    const wrapper = mount(Version, {
      props: {
        ...DEFAULT_PROPS,
      },
    });

    const badgeEl = wrapper.find(`[data-testid="ui-badge"]`);
    const versionText = badgeEl.find(`[data-testid="ui-badge-value"]`).text();

    expect(versionText).toBe(DEFAULT_PROPS.version);
    expect(versionText).toMatchSnapshot();
  });
});
