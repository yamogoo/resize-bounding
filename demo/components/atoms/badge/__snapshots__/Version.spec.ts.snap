// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Version > should display "version" (0.0.1) 1`] = `"0.0.1"`;
